pkg/compose/down.go

```diff
@@ -190,16 +190,14 @@ func (s *composeService) removeNetwork(ctx context.Context, name string, w progr
 	return nil
 }
 
-func (s *composeService) getServiceImages(options api.DownOptions, project *types.Project) map[string]struct{} {
+func (s *composeService) getServiceImagesToRemove(options api.DownOptions, project *types.Project) map[string]struct{} {
 	images := map[string]struct{}{}
 	for _, service := range project.Services {
-		image := service.Image
+		image := service.Labels[api.ImageNameLabel] // Information on the compose file at the creation of the container
 		if options.Images == "local" && image != "" {
 			continue
 		}
-		if image == "" {
-			image = api.GetImageNameOrDefault(service, project.Name)
-		}
+		image = api.GetImageNameOrDefault(service, project.Name)
```

Comment:
I think the changes in this function should be reverted -- the `projectFromName` now makes sure the project loads consistently, so `service.Image` should be `""` here, right?

Otherwise, as it stands right now, if you `compose down --rmi=local` on a project that was launched with previous version of Compose, this will behave like `--rmi=all`. (I just confirmed/tested this with an Awesome Compose sample, launched with v2.9.0, then ran `down --rmi=local` with this PR and it removed the `mariadb:10-focal` image 🙃)

Reply:
`service.Image` will always have something in it, since containers need an image to be created. 
That's the reason why I needed to add a new label to express the content of the `service::image` field when building the image.

Reply:
So `container.Image` will always have a value, which is what the label is for, so the label (which should be `""`) will be used to overwrite `service.Image` @ https://github.com/docker/compose/pull/9715/files#diff-5dabf34848ad18b1b30364b2fd814b7678dbcc4d398227392578ef8266b0ac67R139

Repro example:
**`compose.yaml`**
```yaml
services:
  nginx:
    image: nginx
```

**Current Behavior**
```shell
$ docker-compose-2.9.0 up --wait
[+] Running 2/2
 ⠿ Network rmi_default    Created
 ⠿ Container rmi-nginx-1  Healthy

$ docker-compose-2.9.0 down --rmi=local
[+] Running 2/1
 ⠿ Container rmi-nginx-1  Removed
 ⠿ Network rmi_default    Removed
```

**`up` On 2.9.0, `down` on PR#9715**
```shell
$ docker-compose-2.9.0 up --wait
[+] Running 2/2
 ⠿ Network rmi_default    Created
 ⠿ Container rmi-nginx-1  Healthy

$ docker-compose-9715 down --rmi=local
[+] Running 3/1
 ⠿ Container rmi-nginx-1  Removed
 ⠿ Image nginx            Removed  # <-- 🚨🙀
 ⠿ Network rmi_default    Removed
```

Reply:
Nice catch! 
Just fixed.